### PR TITLE
feat: add `rating` helm chart

### PR DESCRIPTION
**Pull request by jplayout**
This PR is associated with the issue [#1627](https://github.com/jenkins-infra/helpdesk/issues/1627)
I started from the basic "helm chart" and the PR from halkeye here: https://github.com/jenkins-infra/kubernetes-management/pull/137/files
I tested it on my local kubernetes with postgresql and I have the "403" error return page as expected

### CODE REVIEW

File: charts/rating/templates/secret.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  labels:
+{{ include "rating.labels" . | indent 4 }}
+stringData:
+  dbconfig.php: |
+    <?php
+    $dbuser={{ .Values.database.username | quote }};
+    $dbpass={{ .Values.database.password | quote }};
+    $dbname={{ .Values.database.name | quote }};
+    $dbserver={{ .Values.database.server | quote }};
+    $dbport={{ .Values.database.port | toString | quote }};
```
Comment (lemeurherve) [inline charts/rating/templates/secret.yaml:14]:
```suggestion
    $dbport={{ .Values.database.port | toString | quote }};

```

Comment (lemeurherve) [reply charts/rating/templates/secret.yaml:14]: Files have to finish by a new line.
Could you add one to the other too?

**Comment by lemeurherve** [comment]
Look good, congrats! I'll take a look at it tomorrow :)

File: charts/rating/templates/service.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+spec:
+  type: {{ .Values.service.type }}
+  ports:
+    - port: {{ .Values.service.port }}
+      targetPort: http
+      protocol: TCP
+      name: http
+  selector:
+    app.kubernetes.io/name: {{ include "rating.name" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
```
Comment (lemeurherve) [inline charts/rating/templates/service.yaml:16]:
```suggestion
    app.kubernetes.io/instance: {{ .Release.Name }}

```

File: charts/rating/templates/serviceaccount.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{{- if .Values.serviceAccount.create -}}
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ template "rating.serviceAccountName" . }}
+  labels:
+{{ include "rating.labels" . | indent 4 }}
+{{- end -}}
```
Comment (lemeurherve) [inline charts/rating/templates/serviceaccount.yaml:8]:
```suggestion
{{- end -}}

```

File: charts/rating/templates/tests/test-connection.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+{{ include "rating.labels" . | indent 4 }}
+  annotations:
+    "helm.sh/hook": test-success
+spec:
+  containers:
+    - name: wget
+      image: busybox
+      command: ['wget']
+      args:  ['{{ include "rating.fullname" . }}:{{ .Values.service.port }}']
+  restartPolicy: Never
```
Comment (lemeurherve) [inline charts/rating/templates/tests/test-connection.yaml:15]:
```suggestion
  restartPolicy: Never

```

File: charts/rating/values.yaml [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+  # runAsNonRoot: true
+  # runAsUser: 1000
+
+service:
+  type: ClusterIP
+  port: 80
+
+# Please define ingress settings into environment variable
+ingress:
+  enabled: true
```
Comment (lemeurherve) [inline charts/rating/values.yaml:40]:
```suggestion
  enabled: false
```
I think it's better to disable the exposition on the internet by default, and let people enabling it when they decide to.

**Review by lemeurherve — APPROVED** [review]
Looks good to me :)

Will see next week how to deal with the database, if we'll reuse the existing one or if we'll need a new one inside the cluster.

File: charts/rating/values.yaml (hunk 5 above)
Comment (jplayout) [reply charts/rating/values.yaml:40]: I don't know how you overload your configuration so it's up to you

File: charts/rating/values.yaml [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+#    - host: rating.jenkins.io
+#      paths:
+#        - path: /
+#
+#  tls:
+#    - secretName: rating-tls
+#      hosts:
+#        - rating.jenkins.io
+
+
```
Comment (lemeurherve) [inline charts/rating/values.yaml:56]:
```suggestion

```

**Review by lemeurherve — APPROVED** [review]

**Pull request merged**